app.js

```diff
@@ -422,32 +421,16 @@ module.exports = {
       // Validate the DB exists, before proceeding to restore
       .then((restoreDbClient) => validateRestoreDb(restoreDbClient))
       .then((restoreDbClient) => {
-        const output = new Writable({
-          objectMode: true,
-          write: (restoreBatch, encoding, cb) => {
-            debug(' restored ', restoreBatch.total);
-            // order not guaranteed
-            if (total < restoreBatch.total) total = restoreBatch.total;
-            try {
-              ee.emit('restored', restoreBatch);
-            } finally {
-              cb();
-            }
-          },
-          final: (cb) => {
-            debug('restore complete');
-            ee.emit('finished', { total });
-            cb();
-          }
-        });
-
         return restoreInternal(
           restoreDbClient,
           opts,
           srcStream,
-          output);
+          ee);
+      })
+      .then((total) => {
+        ee.emit('finished', total);
+        callback(null, total);
       })
-      .then(() => { callback(null, { total }); })
       .catch(callback);
```

Comment:
do we trust the whole validator's + restore chain to return converted error?

Reply:
it needs rebasing on the changes from #671 , then the conversion happens here